test/combination_nullable_calc_test.exs

```diff
@@ -0,0 +1,196 @@
+defmodule AshPostgres.CombinationNullableCalcTest do
+  @moduledoc false
+  use AshPostgres.RepoCase, async: false
+  alias AshPostgres.Test.{Post, Author}
+
+  require Ash.Query
+  import Ash.Expr
+
+  describe "combination_of with nullable calculations" do
+    test "combination query with allow_nil? calculation loses ORDER BY" do
+      Post
+      |> Ash.Changeset.for_create(:create, %{title: "Zebra", score: 5})
+      |> Ash.create!()
+
+      Post
+      |> Ash.Changeset.for_create(:create, %{title: "Apple", score: 25})
+      |> Ash.create!()
+
+      Post
+      |> Ash.Changeset.for_create(:create, %{title: "Dog", score: 10})
+      |> Ash.create!()
+
+      Post
+      |> Ash.Changeset.for_create(:create, %{title: "Cat", score: 20})
+      |> Ash.create!()
+
+      query =
+        Post
+        |> Ash.Query.sort([{:title, :asc}])
+        |> Ash.Query.load([:latest_comment_title])
+        |> Ash.Query.combination_of([
+          Ash.Query.Combination.base(
+            filter: expr(score < 15),
+            calculations: %{
+              sort_order: calc(score * 20, type: :integer)
+            },
+            sort: [{calc(score * 20, type: :integer), :desc}]
+          ),
+          Ash.Query.Combination.union(
+            filter: expr(score >= 15),
+            calculations: %{
+              sort_order: calc(score * 5, type: :integer)
+            },
+            sort: [{calc(score * 5, type: :integer), :desc}]
+          )
+        ])
+        |> Ash.Query.sort([{calc(^combinations(:sort_order)), :desc}], prepend?: true)
+
+      result = Ash.read!(query)
+      titles = Enum.map(result, & &1.title)
+      # Expected order: sort_order DESC, then title ASC
+      # Dog(200), Apple(125), Cat(100), Zebra(100)
+      expected_title_order = ["Dog", "Apple", "Cat", "Zebra"]
+      assert titles == expected_title_order
+    end
```

Comment:
This test appears to be failing on an ecto error.

Reply:
That may have been a local glitch.  Is now failing locally with an incorrect order.